src/Letter.ts

```diff
@@ -0,0 +1,119 @@
+import type { TDocumentDefinitions, ContentText, ContentImage, Content } from 'pdfmake/interfaces';
+import { layout as dinLayoutFunction } from './layouts/din-5008-a';
+import { formatAddress, mm2pt, stripTags, applyFunctionToPdfMakeContent } from './utility';
+import type { LayoutFunction, PdfMakeContent, Signature } from './types';
+
+type LetterProps = {
+    sender_address: string[] | string;
+    recipient_address: string[] | string;
+    /** The information block usually displayed at the top-right of the letter. */
+    information_block?: PdfMakeContent;
+    subject: string;
+    content: PdfMakeContent;
+    signature: Signature;
+};
+// The `marginTop` is missing in the type definitions.
+type PdfMakeSignature = (ContentText | ContentImage) & { marginTop: number };
+
+const _parseTags = (content: string) => {
+    if (!content) return [];
+
+    const regex = /<(.+?>)([\S\s]*?)<\/\1/gmu;
+    const text_array = content.split(regex);
+
+    const content_array: Content[] = [];
+    text_array.forEach(function (slice, i) {
+        switch (slice) {
+            case 'bold>':
+                content_array.push({ text: text_array[i + 1], bold: true });
+                delete text_array[i + 1];
+                break;
+            case 'italic>':
+                content_array.push({ text: text_array[i + 1], italics: true });
+                delete text_array[i + 1];
+                break;
+            case '':
+                break;
+            default:
+                content_array.push(slice);
+        }
+    });
+
+    return content_array;
+};
+
+export class Letter {
+    doc: TDocumentDefinitions;
+    layout: LayoutFunction;
+    props: LetterProps;
+
+    constructor(props: LetterProps, layout_function?: LayoutFunction) {
+        this.doc = { content: '' };
+
+        this.layout = layout_function || dinLayoutFunction;
+        this.props = {
+            sender_address: [],
+            recipient_address: [],
+            information_block: '',
+            subject: '',
+            content: '',
+            signature: { type: 'text', name: '' },
+        };
+        this.setProps(props);
+    }
+
+    /** Render the letter as a string, used e.g. for emails. */
+    toString() {
```

Comment:
This also (even though it is obvious).

Reply:
Same here. TS infers `string` just fine. I'm not a big fan of being unnecessarily explicit with types when TS can infer them correctly.